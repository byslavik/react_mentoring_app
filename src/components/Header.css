.header {
    margin: 0 auto;
    max-width: 1000px;
}

.header nav {
    margin: 10px 0;
}
.header .siteHeader {
    font-size: 80px;
    text-align: center;
    text-transform: uppercase;
    color: #fff;
    text-shadow: 0px 3px 15px #000;
}
.header nav li {
    margin: 0;
    display: inline-block;
    margin-right: 10px;
    padding: 0;
    list-style-type: none;
}
.headerWrapper {
    min-height: 220px;
    background: url('/public/img/headerBg.jpg');
    padding: 10px;
    background-size: cover;
}