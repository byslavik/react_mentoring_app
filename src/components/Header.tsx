import * as React from 'react';
import * as s from './Header.scss';

interface HeaderState {
}
interface HeaderProps {
}

export class Header extends React.Component<HeaderProps, HeaderState> {
  
    render() {
        return (<div className={s.headerWrapper}>
            <header className={s.header}>
                {this.props.children}
            </header>
        </div>);
    }
}