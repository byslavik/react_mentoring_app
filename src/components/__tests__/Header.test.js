import * as React from 'react';
import {Header} from '../Header';
import * as renderer from 'react-test-renderer';

const ChildrenComponent = ()=> {
  return 'ChildrenComponent';
}

describe('Header component', ()=>{
  it('Header component shows correctly', () => {
    const tree = renderer.create(
      <Header>
        <ChildrenComponent/>
      </Header>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
})