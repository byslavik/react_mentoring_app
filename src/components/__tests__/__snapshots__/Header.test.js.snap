// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Header component shows correctly 1`] = `
<div
  className={undefined}
>
  <header
    className={undefined}
  >
    ChildrenComponent
  </header>
</div>
`;
